ui/common/src/socket.ts

```diff
@@ -289,57 +283,65 @@ export default class StrongSocket {
     const ws = this.ws;
     if (ws) {
       this.debug('Disconnect');
-      this.autoReconnect = false;
       ws.onerror = ws.onclose = ws.onopen = ws.onmessage = () => {};
       ws.close();
     }
   };
 
-  onError = (e: unknown): void => {
+  private onError = (e: unknown): void => {
+    if (this.heartbeat.wasSuspended) return;
     this.options.debug = true;
     this.debug(`error: ${e} ${JSON.stringify(e)}`); // e not always from lila
   };
 
-  onClose = (e: CloseEvent, url: string): void => {
+  private onClose = (e: CloseEvent): void => {
     site.pubsub.emit('socket.close');
-    if (this.autoReconnect) {
+
+    if (this.heartbeat.wasSuspended) return this.onSuspended();
```

Comment:
in order to have valid data, it is critical that we can distinguish task queue suspensions from real network/application errors